test/parquet_schema_test.cc

```diff
@@ -50,17 +136,375 @@ TEST(HasFieldIds, PrimitiveNode) {
 TEST(HasFieldIds, GroupNode) {
   auto group_node_without_field_id =
       MakeGroupNode("test_group", {MakeInt32Node("c1"), MakeInt32Node("c2")});
-  EXPECT_FALSE(HasFieldIds(group_node_without_field_id));
+  EXPECT_FALSE(HasFieldIds(group_node_without_field_id));  // NOLINT
 
   auto group_node_with_full_field_id = MakeGroupNode(
       "test_group",
       {MakeInt32Node("c1", /*field_id=*/2), MakeInt32Node("c2", /*field_id=*/3)},
       /*field_id=*/1);
-  EXPECT_TRUE(HasFieldIds(group_node_with_full_field_id));
+  EXPECT_TRUE(HasFieldIds(group_node_with_full_field_id));  // NOLINT
 
   auto group_node_with_partial_field_id = MakeGroupNode(
       "test_group", {MakeInt32Node("c1", /*field_id=*/1), MakeInt32Node("c2")});
-  EXPECT_TRUE(HasFieldIds(group_node_with_partial_field_id));
+  EXPECT_TRUE(HasFieldIds(group_node_with_partial_field_id));  // NOLINT
```

Comment:
nits: could we add a comment saying why there's a nolint (and/or use `NOLINT(name-of-lint)` to be specific), and maybe NOLINTBEGIN/NOLINTEND would be cleaner (up to you)?